cmd/cli/plugin/feature/activate_test.go

```diff
@@ -0,0 +1,258 @@
+
+	corev1alpha2 "github.com/vmware-tanzu/tanzu-framework/apis/core/v1alpha2"
+	"github.com/vmware-tanzu/tanzu-framework/featuregates/client/pkg/featuregateclient"
+	"github.com/vmware-tanzu/tanzu-framework/featuregates/client/pkg/featuregateclient/fake"
+)
+
+func TestActivateFeature(t *testing.T) {
+	type nullBool struct {
+		wasSet bool
+		setTo  bool
+	}
+
+	tests := []struct {
+		description  string
+		featureName  string
+		userAllows   nullBool
+		wantErr      error
+		wantActivate bool
+	}{
+		{
+			description:  "activate a technical preview feature",
+			featureName:  "tuna",
+			wantErr:      nil,
+			userAllows:   nullBool{wasSet: false},
+			wantActivate: true,
+		},
+		{
+			description:  "activate a deprecated feature",
+			featureName:  "biz",
+			wantErr:      nil,
+			userAllows:   nullBool{wasSet: false},
+			wantActivate: true,
+		},
+		{
+			description:  "cannot activate a feature that was not found in cluster",
+			featureName:  "hard-to-get",
+			userAllows:   nullBool{wasSet: false},
+			wantErr:      featuregateclient.ErrTypeNotFound,
+			wantActivate: false,
+		},
+		{
+			description:  "activate a stable feature that is already activated",
+			featureName:  "super-toaster",
+			userAllows:   nullBool{wasSet: false},
+			wantErr:      nil,
+			wantActivate: true,
+		},
+		{
+			description:  "cannot activate a feature that is not gated by feature gate",
+			featureName:  "specialized-toaster",
+			userAllows:   nullBool{wasSet: false},
+			wantErr:      featuregateclient.ErrTypeNotFound,
+			wantActivate: false,
+		},
+		{
+			description:  "cannot activate a feature that is gated by more than one feature gate",
+			featureName:  "baz",
+			userAllows:   nullBool{wasSet: false},
+			wantErr:      featuregateclient.ErrTypeTooMany,
+			wantActivate: false,
+		},
+		{
+			description:  "activate experimental feature with user permission",
+			featureName:  "cloud-event-speaker",
+			userAllows:   nullBool{wasSet: true, setTo: true},
+			wantErr:      nil,
+			wantActivate: true,
+		},
+		{
+			description:  "don't activate experimental feature when user disallows it",
+			featureName:  "cloud-event-relayer",
+			userAllows:   nullBool{wasSet: true, setTo: false},
+			wantErr:      featuregateclient.ErrTypeForbidden,
+			wantActivate: false,
+		},
+		{
+			description:  "don't activate experimental feature when user does not give permission",
+			featureName:  "cloud-event-relayer",
+			userAllows:   nullBool{wasSet: false},
+			wantErr:      featuregateclient.ErrTypeForbidden,
+			wantActivate: false,
+		},
+	}
+
+	objs, _, _ := fake.GetTestObjects()
+	s := scheme.Scheme
+	if err := corev1alpha2.AddToScheme(s); err != nil {
+		t.Fatalf("add config scheme: (%v)", err)
+	}
+
+	cl := crclient.NewClientBuilder().WithRuntimeObjects(objs...).Build()
+	fgClient, err := featuregateclient.NewFeatureGateClient(featuregateclient.WithClient(cl))
+	if err != nil {
+		t.Fatalf("get FeatureGate client: %v", err)
```

Comment:
```suggestion
		t.Fatalf("could not get FeatureGate client: %v", err)
```

Reply:
can you make this change in other places of this pr as well?

Reply:
Done.